Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -2596,4 +2638,115 @@ void SoftwareGLContext::present()
+            break;
+        case GL_LINEAR:
+            config.texture_min_filter = SoftGPU::TextureFilter::Linear;
+            config.mipmap_filter = SoftGPU::MipMapFilter::None;
+            break;
+        case GL_NEAREST_MIPMAP_NEAREST:
+            config.texture_min_filter = SoftGPU::TextureFilter::Nearest;
+            config.mipmap_filter = SoftGPU::MipMapFilter::Nearest;
+            break;
+        case GL_NEAREST_MIPMAP_LINEAR:
+            config.texture_min_filter = SoftGPU::TextureFilter::Linear;
+            config.mipmap_filter = SoftGPU::MipMapFilter::Nearest;
+            break;
+        case GL_LINEAR_MIPMAP_LINEAR:
+            config.texture_min_filter = SoftGPU::TextureFilter::Linear;
+            config.mipmap_filter = SoftGPU::MipMapFilter::Linear;
+            break;
+        }
+
+        switch (sampler.mag_filter()) {
+        case GL_NEAREST:
+            config.texture_mag_filter = SoftGPU::TextureFilter::Nearest;
+            break;
+        case GL_LINEAR:
+            config.texture_mag_filter = SoftGPU::TextureFilter::Linear;
+            break;
+        }
+
+        switch (sampler.wrap_s_mode()) {
+        case GL_CLAMP:
+            config.texture_wrap_u = SoftGPU::TextureWrapMode::Clamp;
+            break;
+        case GL_CLAMP_TO_BORDER:
+            config.texture_wrap_u = SoftGPU::TextureWrapMode::ClampToBorder;
+            break;
+        case GL_CLAMP_TO_EDGE:
+            config.texture_wrap_u = SoftGPU::TextureWrapMode::ClampToEdge;
+            break;
+        case GL_REPEAT:
+            config.texture_wrap_u = SoftGPU::TextureWrapMode::Repeat;
+            break;
+        case GL_MIRRORED_REPEAT:
+            config.texture_wrap_u = SoftGPU::TextureWrapMode::MirroredRepeat;
+            break;
+        }
+
+        switch (sampler.wrap_t_mode()) {
+        case GL_CLAMP:
+            config.texture_wrap_v = SoftGPU::TextureWrapMode::Clamp;
+            break;
+        case GL_CLAMP_TO_BORDER:
+            config.texture_wrap_v = SoftGPU::TextureWrapMode::ClampToBorder;
+            break;
+        case GL_CLAMP_TO_EDGE:
+            config.texture_wrap_v = SoftGPU::TextureWrapMode::ClampToEdge;
+            break;
+        case GL_REPEAT:
+            config.texture_wrap_v = SoftGPU::TextureWrapMode::Repeat;
+            break;
+        case GL_MIRRORED_REPEAT:
+            config.texture_wrap_v = SoftGPU::TextureWrapMode::MirroredRepeat;
+            break;
+        }
+
+        switch (m_texture_units[i].env_mode()) {
+        case GL_MODULATE:
+            config.fixed_function_texture_env_mode = SoftGPU::TextureEnvMode::Modulate;
+            break;
+        case GL_REPLACE:
+            config.fixed_function_texture_env_mode = SoftGPU::TextureEnvMode::Replace;
+            break;
+        case GL_DECAL:
+            config.fixed_function_texture_env_mode = SoftGPU::TextureEnvMode::Decal;
+            break;
+        default:
+            VERIFY_NOT_REACHED();
```

Comment:
To be consistent, the other `switch` blocks above should also get a default section - it looks weird to have it only here.